ext/date/php_date.c

```diff
@@ -4389,14 +4389,16 @@ static int php_date_interval_initialize_from_hash(zval **return_value, php_inter
 	PHP_DATE_INTERVAL_READ_PROPERTY("h", h, timelib_sll, -1)
 	PHP_DATE_INTERVAL_READ_PROPERTY("i", i, timelib_sll, -1)
 	PHP_DATE_INTERVAL_READ_PROPERTY("s", s, timelib_sll, -1)
-	do {
+	{
 		zval *z_arg = zend_hash_str_find(myht, "f", sizeof("f") - 1);
+		(*intobj)->diff->us = -1000000;
```

Comment:
How did you pick this value? I see some other code in this file using `-99999`.  @derickr What's the correct value for an unset `us`?

Reply:
Okay, it's clear how you picked it: Used in the existing code two lines below... Still, I'm not sure this is the right value, as TIMELIB_UNSET is -99999.

Reply:
Hmm, now I'm confused, since the other properties seem to use `-1` as unset default? @derickr, please clarify. Thanks.

Reply:
I'm going to look into this when I return from Christmas holiday. Ping me the first week of Jan if I haven't replied yet

Reply:
Ping ;)

Reply:
``-1000000`` is correct, as in these lines:
```
4220         if (strcmp(Z_STRVAL_P(member), "f") == 0) {
4221             fvalue = obj->diff->us / 1000000.0;
4222             break;
4223         }
```

and

```
4238     if (fvalue != -1) {
4239         ZVAL_DOUBLE(retval, fvalue);
```

The `fvalue` ends up being `-1` during the division, which would result in FALSE being returned.